src/extension.ts

```diff
@@ -331,7 +342,39 @@ function launchMetals(
           const panel = getDoctorPanel(isReload);
           panel.webview.html = html;
         }
+      } else {
+        switch (params.command) {
+          case "metals-goto-location":
+            const location =
+              params.arguments && (params.arguments[0] as Location);
```

Comment:
I really don't like that we're casting at use site, but this is not strictly related to this PR.

Ideally, we should validate at the networking layer and use well-typed models after that. We do this at work using io-ts, I can experiment with it sometime in the near future.

It's 👍 for this PR.

Reply:
Validating the input here would be nice  and would benefit the other client commands that also cast 👍